semantics/c/language/common/memory/writing.k

```diff
@@ -75,7 +75,7 @@ module C-MEMORY-WRITING
           [structural]
      // Write global, shared data, or "allocated" duration.
      rule write(Loc:SymLoc, V:CValue, T:Type) => writeSpecific(Loc, V, T, false)
-          requires (isStaticDuration(Loc) orBool isAllocatedDuration(Loc))
+          requires (isStaticDuration(Loc) orBool isAllocatedDuration(Loc) orBool isHardwareLoc(Loc))
```

Comment:
this seems weird. wouldn't all hardware locations be static storage? thus, you ought to just modify the isStaticDuration function.

Reply:
Done, I think